### ISSUE TITLE
Knitted help files with evaluated examples

### ISSUE THREAD

**Issue by grantmcdermott** [rstudio/rstudio#10017]
A limitation of R's documentation standards is that examples in most help files don't provide output. For instance, if I type in `?contour` and scroll down to the Examples section at the bottom, then I see a lot of commands and verbal descriptions.... but no visual evidence of the plots themselves. 

![a1](https://user-images.githubusercontent.com/10285037/139150049-62772b8e-4f26-42e1-a6ff-0716ed442ebf.png)

Now, I _could_ use the `example("contour")` function to actually run these examples interactively and view the resulting plots. But the downside to this approach is that it pollutes my global environment with a load of objects that I then need to clean up manually (or, may even have overwritten objects that I had created beforehand).

![a3](https://user-images.githubusercontent.com/10285037/139150047-32266376-6596-4b1f-91d9-ea33cc7a4347.png)

It would be great to have some kind of functionality that runs the `example(<function>)` walkthrough in a sandboxed environment. Even better, would be for everything to get passed through to **knitr** and evaluated. This way, the entire help file could be viewed as as if it were a standard (HTML-knitted) R Markdown file with code and the results embedded in the same document. 

The reason I regard this as a potential RStudio FR &mdash; rather than a knitr and rmarkdown one &mdash; is that this functionality could be controlled with a global IDE option. Typing in, say, `?contour` would then automatically bring up the knitted help file by default.

Thanks for considering.

**Comment by ronblum** [comment]
@grantmcdermott Thank you for this suggestion! We'll review this as we continue development of RStudio. To help us with considering this, what part of `rmdocs` fulfills what you're looking for, and what is the part that's missing?

**Comment by grantmcdermott** [comment]
Hi @ronblum. Quite embarrassed that I dropped the ball on this issue (too many other things going on). 

However, I was just reminded of this after seeing the excellent news that R 4.2.0 will introduce an option to run examples in a browser. https://twitter.com/henrikbengtsson/status/1507034533136269312

For RStudio, I assume that this will automatically default to running inside the Viewer pane. As such, I think we can close.

**Comment by ronblum** [comment]
@grantmcdermott Thank you for the update! I just tried it out with R 4.2.0 alpha in Windows 11, and if `knitr` is installed:
1) The examples are generated as a page in the Help Pane
2) The objects aren't in the global environment
So, I think this accomplishes what you're looking for!

By the way, no need to be embarrassed. If I gathered together all of the balls I've dropped here, there'd be enough for a whole ball pit.
